### PR TITLE
add comments column to lab observation

### LINKED ISSUE #1593: Given a lab result observation, display lab comments
## Action Requested
Add a new column for Lab Comments in Lab Results table. If Lab Comments are present for a given row, the user will be able to click "View Comment" to see the full comment

## Acceptance Criteria
(not sure)

## Testing
(not sure)

## Background & Context
I didn't realize Lab Comments were associated with each row in a results table, so the field needed to be redesigned given this new info

## Designs
[Figma Designs](https://www.figma.com/file/tXWUeV3Kjehd7VNqHojZYq/eCR-Viewer?type=design&node-id=2771%3A43987&mode=design&t=krtjxIZYAAAL6epd-1)
[Figma Clickable Prototype](https://www.figma.com/proto/tXWUeV3Kjehd7VNqHojZYq/eCR-Viewer?page-id=2771%3A43987&type=design&node-id=2771-43988&viewport=647%2C4421%2C0.43&t=6Pn2HJ58Sf8ivU9m-1&scaling=min-zoom&starting-point-node-id=2771%3A43988&mode=design)

Collapsed (Default):

Expanded:

### PR DESCRIPTION
**Pull request by BobanL**
# PULL REQUEST

## Summary
- Add new column for lab observation

## Related Issue
Fixes #1593 

## Additional Information

https://github.com/CDCgov/phdi/assets/10108172/b742a81d-014f-4893-8ff2-831ad1eecd78

### CODE REVIEW

**Comment by BobanL** [comment]
@sarahtress Ready for review!

**Comment by sarahtress** [comment]
> @sarahtress Ready for review! image
> 
> image

@BobanL thanks for the screenshots! so this looks worse in table form than I anticipated when making the design initially 😅 totally my fault, but could you try making the line above the comment gray when it's expanded? I just updated the designs with what I have in mind, and here's a screenshot too for reference:

Screenshot 2024-04-19 at 5 34 56 PM

right now it's just hard to tell visually that the comment is associated with the row above it, hopefully this should help!

**Comment by BobanL** [comment]
> right now it's just hard to tell visually that the comment is associated with the row above it, hopefully this should help!

@sarahtress I've made the change - Let me know if this matches! 

image

**Comment by sarahtress** [comment]
Looks good to me thanks!!

*Sarah Tress (she/her)*
Product Designer | skylight.digital

File: containers/ecr-viewer/src/app/evaluate-service.tsx [hunk 1]
```diff
@@ -75,6 +72,70 @@ export const evaluateTable = (
...
+const BuildRow: React.FC<BuildRowProps> = ({
+  columns,
+  mappings,
+  entry,
+}: BuildRowProps) => {
+  const [hiddenComment, setHiddenComment] = useState(true);
+
+  let hiddenRows: React.JSX.Element[] = [];
+  let rowCells = columns.map((column, index) => {
+    let rowCellData: any;
```
Comment (nickbristow) [inline containers/ecr-viewer/src/app/evaluate-service.tsx:92]: we don't want to call this a JSX.element or pull this if chain into a seperate function so we can not make an empty var?

**Review by nickbristow — APPROVED** [review]

**Pull request merged**